### PR TITLE
Updated to nf90 interface, removed #includes

### PR DESCRIPTION
**Pull request by hellkite500**
For reference, some regex magic for nf90 updates (vim regex listed below.) To apply to entire file in vim, prefix each regex with %, i.e. `%s#nf_#nf90#`

Move all nf calls to nf90
```
s#nf_#nf90_#
s#NF_#NF90_#
```

Clean up some other function calls:

Remove 4th arg to function def var (for vim .{-} non greedy search):
```
s#\vnf90_def_var\(\s*(\w{-})\s*,\s*(%(\w|["'/\(\)]){-})\s*,\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\(.{-}\)|%(\w+)))\s*,\s*(\w{-})\)#nf90_def_var(\1, \2, \3, \5, \6)#
```

Fix put_att_text:
```
s#\vnf90_put_att_text\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\w|["'/\(\)^]){-})\s*,\s*(\w{-})\s*,\s*(%(\w|["'^-/().]|\s){-})\s*\)#nf90_put_att(\1, \2, \3, \5)#
```
Fix put_att_real:
```
s#\vnf90_put_att_real\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\w|["'/\(\)^]){-})\s*,\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(-?\w{-}|%(\w+\(.{-}\))|%(\w+\(.{-}\).{-}))\s*\)#nf90_put_att(\1, \2, \3, \6)#
```

Fix put_att_int:
```
s#\vnf90_put_att_int\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\w|["'/\(\)^]){-})\s*,\s*(\w{-})\s*,\s*(-?\w{-}|%(\w+\(.{-}\)))\s*\)#nf90_put_att(\1, \2, \3, \5)#
```
There seem to be two put_att_int interfaces, so need to run this one too:
```
s#\vnf90_put_att_int\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\w|["'/\(\)^]){-})\s*,\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(-?\w{-}|%(\w+\(.{-}\))|%(\w+\(.{-}\).{-}))\s*\)#nf90_put_att(\1, \2, \3, \6)#
```
Had to manually find put_att_int where args cross multiple lines, remove NF90_INT and 1 from call, change to nf90_put_att
Also manually find put_att_text where args cross multiple lines, remove number (i.e. character string length) arg.

Fix nf_put_vara_type
```
s#\vnf90_put_vara_\w+\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\(.{-}\)|%(\w+)))\s*,\s*(%(\(.{-}\)|%(\w+)))\s*,\s*(\w{-}|%(\w+\(.{-}\)))\s*\)#nf90_put_var(\1, \2, \5, \3, \4)#
```

Fix nf90_inq_dimlen:
```
s#\vnf90_inq_dimlen\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(\w{-})\s*\)#nf90_inquire_dimension(\1, \2, len = \3)#
```

Fix nf90_get_var_real/int/double
```
s#\vnf90_get_var_real#nf90_get_var#i
s#\vnf90_get_var_int#nf90_get_var#i
s#\vnf90_get_var_double#nf90_get_var#i
```

Fix nf90_get_att_int/real/text
```
s#\vnf90_get_att_int#nf90_get_att#i
s#\vnf90_get_att_real#nf90_get_att#i
s#\vnf90_get_att_text#nf90_get_att#i
```
Note the `#i` in the above regex, this ignores case.  Can be applied to all regex if case is mixed.

Fix nf_get_vara_type
```
s#\vnf90_get_vara_\w+\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(%(\(.{-}\)|%(\w+)))\s*,\s*(%(\(.{-}\)|%(\w+)))\s*,\s*(\w{-}|%(\w+\(.{-}\)))\s*\)#nf90_get_var(\1, \2, \5, \3, \4)#
```

Clean up any put var with a count of 1.  It confuses the compiler:
i.e. `nf90_put_var(ncid, varid, resht(i), (/start_pos/), (/1/))`
```
s#\vnf90_put_var\(\s*(\w{-})\s*,\s*(\w{-})\s*,\s*(\w{-}|%(\w+\(.{-}\)))\s*,\s*(%(\(.{-}\)|%(\w+)))\s*,\s*(\(\/1\/\))\s*\)#nf90_put_var(\1, \2, \3, \4)#
```

### CODE REVIEW

File: trunk/NDHMS/MPP/mpp_land.F [hunk 1]
```diff
@@ -22,9 +22,8 @@
 MODULE MODULE_MPP_LAND
 
   use MODULE_CPL_LAND
-
+  use mpi
```
Comment (jmccreight) [inline trunk/NDHMS/MPP/mpp_land.F]: I dont understand this one

Comment (jmccreight) [reply trunk/NDHMS/MPP/mpp_land.F]: Ah, I see the removed include statement now.

File: trunk/NDHMS/Routing/module_HYDRO_io.F [hunk 2]
```diff
@@ -2943,30 +2931,30 @@ subroutine sub_output_gw(igrid, split_output_count, ixrt, jxrt, nsoil, &
...
+     iret = nf90_put_var(ncid, varid, xdumd, (/1,1,output_count/), (/ixrtd,jxrtd,1/))
      
 #ifdef MPP_LAND
         xdumd = gQgw_chanrt
 #else  
         xdumd = qgw_chanrt
 #endif
 
-     iret = nf_inq_varid(ncid,  "QGW_CHANRT", varid)
-     iret = nf_put_vara_real(ncid, varid, (/1,1,output_count/), (/ixrtd,jxrtd,1/), xdumd)
```
Comment (jmccreight) [inline trunk/NDHMS/Routing/module_HYDRO_io.F]: This pattern is not trivial to replace and happens frequently. Just seems worth noting.

Comment (hellkite500) [reply trunk/NDHMS/Routing/module_HYDRO_io.F]: What do you mean not trivial to replace?

**Pull request merged**